test/sql/constraint.sql

```diff
@@ -0,0 +1,299 @@
+\o /dev/null
+\ir include/create_single_db.sql
+\o
+
+CREATE TABLE hyper (
+  time BIGINT NOT NULL,
+  device_id TEXT NOT NULL,
+  sensor_1 NUMERIC NULL DEFAULT 1 CHECK (sensor_1 > 10)
+);
+
+
+SELECT * FROM create_hypertable('hyper', 'time', chunk_time_interval => 10);
+
+--check and not-null constraints are inherited through regular inheritance.
+
+\set ON_ERROR_STOP 0
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 9);
+
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, NULL, 11);
+
+\set ON_ERROR_STOP 1
+
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+
+INSERT INTO hyper(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+
+----------------------- UNIQUE CONSTRAINTS ------------------
+
+CREATE TABLE hyper_unique (
+  time BIGINT NOT NULL UNIQUE,
+  device_id TEXT NOT NULL,
+  sensor_1 NUMERIC NULL DEFAULT 1 CHECK (sensor_1 > 10)
+);
+
+
+SELECT * FROM create_hypertable('hyper_unique', 'time', chunk_time_interval => 10);
+
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987800000000000, 'dev2', 11);
+
+
+\set ON_ERROR_STOP 0
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev2', 11);
+\set ON_ERROR_STOP 1
+
+\d+ hyper
+\d+ hyper_unique
+--should have unique constraint not just unique index
+\d+ _timescaledb_internal._hyper_2_4_chunk
+
+ALTER TABLE hyper_unique DROP CONSTRAINT hyper_unique_time_key;
+\d+ _timescaledb_internal._hyper_2_4_chunk
+
+--uniqueness not enforced
+INSERT INTO hyper_unique(time, device_id,sensor_1) VALUES
+(1257987700000000000, 'dev3', 11);
+
+
+--shouldn't be able to create constraint
+\set ON_ERROR_STOP 0
```

Comment:
Do we need this around every test case? Seems somewhat unnecessary and makes the tests much longer. If you set this top level or group error cases together, the test would be a lot more clear. We should be able to verify error cases in the test output anyway so this is only ensuring hard stop.

Reply:
I think I prefer the extra safety of doing it this way. Do you feel strongly?